### PR TITLE
Added build verification workflow for all tools

### LINKED ISSUE #109: Add build verification workflow for all tools
Add build verification workflow for all tools

---
[Associated WorkItem - 59877](https://dev.azure.com/msft-skilling/Content/_workitems/edit/59877)

### PR DESCRIPTION
**Pull request by IEvangelist**
In this PR:

- Added _dotnet-build-verification.yml_ GitHub Action workflow: builds and tests .NET bits.
- Added _node-build-verification.yml_ GitHub Action workflow: builds and tests NodeJS bits.
- Upgraded everything to .NET 7 and NodeJS 18.
- Upgrade all package references, both for .NET and Node.
- Attempted to fix broken `FakeGitHubClient` for tests, fixed one use case, and applied `Fact(Skip = "...")` to the other two that were failing.
- In TypeScript code, use properly scoped imports, i.e.; instead of importing all, use explicit imports to minimize the overall size.
- Ran `npm run lint` and `npm run format` commands.
- Refactored all namespaces to be `file-scoped`.

Fixes #109

### CODE REVIEW

File: actions/docs-verifier/src/ActionRunner/ActionRunner.csproj [hunk 1]
```diff
@@ -2,12 +2,19 @@
...
+    <TargetFramework>net7.0</TargetFramework>
   </PropertyGroup>
 
   <ItemGroup>
     <ProjectReference Include="..\MarkdownLinksVerifier\MarkdownLinksVerifier.csproj" />
     <ProjectReference Include="..\RedirectionVerifier\RedirectionVerifier.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="Microsoft.CodeAnalysis.NetAnalyzers" Version="7.0.0">
```
Comment (gewarren) [inline actions/docs-verifier/src/ActionRunner/ActionRunner.csproj:14]: Why do you need to add this reference?

Comment (IEvangelist) [reply actions/docs-verifier/src/ActionRunner/ActionRunner.csproj:14]: Good question, I believe Visual Studio added this on its own. I didn't explicitly add it, but it certainly doesn't hurt anything. I can remove it if you'd like.

Comment (Youssef1313) [reply actions/docs-verifier/src/ActionRunner/ActionRunner.csproj:14]: This should be removed and the version should be updated in https://github.com/dotnet/docs-tools/blob/0c7914887bd21500d2dd4d1994c26ce982f3a237/actions/docs-verifier/Directory.Build.props#L15

Comment (gewarren) [reply actions/docs-verifier/src/ActionRunner/ActionRunner.csproj:14]: It shouldn't be needed even if you are running code analysis. The analyzers are part of the .NET SDK now.

Comment (Youssef1313) [reply actions/docs-verifier/src/ActionRunner/ActionRunner.csproj:14]: Ah, I might have added it to workaround some bug that wasn't yet shipped to analyzers in the .NET SDK.

File: actions/docs-verifier/Directory.Build.props [hunk 2]
```diff
@@ -12,7 +11,6 @@
 
   <ItemGroup>
     <!-- This preview version fixes a CA1812 (Avoid uninstantiated internal classes) false positive bug with top-level statements. -->
```
Comment (Youssef1313) [inline actions/docs-verifier/Directory.Build.props:13]:
```suggestion
```

**Review by BillWagner — APPROVED** [review]
This LGTM.

I would like @adegeo to check the repoman changes though.

**Comment by IEvangelist** [comment]
> This LGTM.
> 
> I would like @adegeo to check the repoman changes though.

The only changes to repoman were file-scoped namespaces, and upgrading of dependent package versions. Nothing else changed.

**Pull request merged**